test/unittest/component/test_graph.py

```diff
@@ -0,0 +1,128 @@
+from test.db_config import DBConfig
+
+import networkx as nx
+import pytest
+
+from superduperdb.components.graph import Graph
+from superduperdb.components.model import Model
+
+
+@pytest.fixture
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def model1(db):
+    def model_object(x):
+        return x + 1
+
+    model = Model(identifier='m1', object=model_object)
+    db.add(model)
+    yield model
+
+
+@pytest.fixture
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def model2(db):
+    def model_object(x):
+        return x + 2
+
+    model = Model(identifier='m2', object=model_object)
+    db.add(model)
+    yield model
+
+
+@pytest.fixture
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def model2_multi(db):
+    def model_object(x, y):
+        return x + y + 2
+
+    model = Model(identifier='m2_multi', object=model_object)
+    db.add(model)
+    yield model
+
+
+@pytest.fixture
+@pytest.mark.parametrize("db", [DBConfig.mongodb_empty], indirect=True)
+def model3(db):
+    def model_object(x, y):
+        return x + y + 3
+
+    model = Model(identifier='m3', object=model_object)
+    db.add(model)
+    yield model
+
+
+def test_simple_graph(model1, model2):
+    g = Graph(identifier='simple-graph')
+    intermediate_node = g.connect(g, model1)
```

Comment:
`g.connect(m.outputs[0], model1.inputs['X']`

Reply:
```python
G = GraphModel(
    nodes=[m1, m2, m3],
    edges=[
        (m1.outputs[0], m2.inputs['X']),
        (m2.outputs[1], m3.inputs['Y'])
    ],
    input=m1.inputs,
    outputs=[*m2.outputs, *m3.outputs],
)

# syntatic sugar:
G.connect(m1.outputs[0], m2.inputs['X'])

# under the hood JSON-able data:
G.add_edge([(m1.identifier, 0), (m2, 'X')])
```

Reply:
Absolute minimum:

```python
G = GraphModel(
    nodes=[m1, m2, m3],
    edges=[
        ((m1.identifier, 0), (m3.identifier, 'X')),
        ((m2.identifier, None), (m3, identifier, 'Y'))
    ],
    input=m1.identifier,
    outputs=[m2.identifier, m3.identifier],
)
```

Reply:
So we are assuming only 1 input node, but potentially multiple output nodes.
The outputs of all output nodes are concatenated into one output list.

Reply:
done